### PR TITLE
Merriam-Webster Dictionary Tool

### LINKED ISSUE #12039: Tools for Dictionary APIs
### Feature request

It would be nice to have agents that could access dictionary APIs such as the Merriam-Webster API or Urban Dictionary API (for slang).

### Motivation

It can be useful to be able to look up definitions for words using a dictionary to provide additional context. With no current dictionary tools available, it would be beneficial for there to be an implemented dictionary tool available at all.

### Your contribution

We will open a PR that adds a new tool for accessing the Merriam-Webster Collegiate Dictionary API (https://dictionaryapi.com/products/api-collegiate-dictionary[/](https://www.dictionaryapi.com/)), which provides definitions for English words, as soon as possible. In the future this could be extended to support other Merriam-Webster APIs such as their Medical Dictionary API (https://dictionaryapi.com/products/api-medical-dictionary) or Spanish-English Dictionary API (https://dictionaryapi.com/products/api-spanish-dictionary).

We may also open another PR for Urban Dictionary API integration.

### PR DESCRIPTION
**Pull request by Zollerboy1**
# Description

We implemented a simple tool for accessing the Merriam-Webster Collegiate Dictionary API (https://dictionaryapi.com/products/api-collegiate-dictionary).

Here's a simple usage example:

```py
from langchain.llms import OpenAI
from langchain.agents import load_tools, initialize_agent, AgentType

llm = OpenAI()
tools = load_tools(["serpapi", "merriam-webster"], llm=llm) # Serp API gives our agent access to Google
agent = initialize_agent(
  tools, llm, agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION, verbose=True
)
agent.run("What is the english word for the german word Himbeere? Define that word.")
```

Sample output:

```
> Entering new AgentExecutor chain...
 I need to find the english word for Himbeere and then get the definition of that word.
Action: Search
Action Input: "English word for Himbeere"
Observation: {'type': 'translation_result'}
Thought: Now I have the english word, I can look up the definition.
Action: MerriamWebster
Action Input: raspberry
Observation: Definitions of 'raspberry':

1. rasp-ber-ry, noun: any of various usually black or red edible berries that are aggregate fruits consisting of numerous small drupes on a fleshy receptacle and that are usually rounder and smaller than the closely related blackberries
2. rasp-ber-ry, noun: a perennial plant (genus Rubus) of the rose family that bears raspberries
3. rasp-ber-ry, noun: a sound of contempt made by protruding the tongue between the lips and expelling air forcibly to produce a vibration; broadly : an expression of disapproval or contempt
4. black raspberry, noun: a raspberry (Rubus occidentalis) of eastern North America that has a purplish-black fruit and is the source of several cultivated varieties —called also blackcap

Thought: I now know the final answer.
Final Answer: Raspberry is an english word for Himbeere and it is defined as any of various usually black or red edible berries that are aggregate fruits consisting of numerous small drupes on a fleshy receptacle and that are usually rounder and smaller than the closely related blackberries.

> Finished chain.
```

# Issue

This closes #12039.

# Dependencies

We added no extra dependencies.

### CODE REVIEW

**Comment by Zollerboy1** [comment]
@baskaryan @eyurtsev @hwchase17 sorry to mention you all three. Are there any news on this?

File: libs/langchain/langchain/utilities/merriam_webster.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+"""Util that calls Merriam-Webster."""
+import json
+from typing import Dict, Iterator, List, Optional
+from urllib.parse import quote
+
+import requests as req
```
Comment (hwchase17) [inline libs/langchain/langchain/utilities/merriam_webster.py:6]: this is a bit unconventional, i think just `import requests` is more than okay

**Comment by hwchase17** [comment]
sorry for the delay @Zollerboy1 - taking a look now!

**Comment by Zollerboy1** [comment]
Hey @hwchase17, thanks for fixing the tests. I guess, I didn't retest after merging master earlier today.

**Pull request merged**